documentation/trunk/solutions/basic-computing/eiffelbase/eiffelbase-tutorial/eiffelbase-kernel.wiki

```diff
@@ -112,13 +111,16 @@ A number of classes offer facilities which are very close to the language level.
 
 ==Basic types==
 
-The basic types [[ref:libraries/base/reference/boolean_chart|BOOLEAN]] , [[ref:libraries/base/reference/character_8_chart|CHARACTER]] , [[ref:libraries/base/reference/integer_32_chart|INTEGER]] , [[ref:libraries/base/reference/real_32_chart|REAL]]  and [[ref:libraries/base/reference/real_64_chart|DOUBLE]]  are defined by classes of the Kernel library. <br/>
-In reading the class specifications for the numeric types [[ref:libraries/base/reference/integer_32_chart|INTEGER]] , [[ref:libraries/base/reference/real_32_chart|REAL]]  and [[ref:libraries/base/reference/real_64_chart|DOUBLE]] , you might think that the type declarations are too restrictive. For example the addition operation in class [[ref:libraries/base/reference/real_32_chart|REAL]]  reads 
+The basic types [[ref:libraries/base/reference/boolean_chart|BOOLEAN]] ,
+[[ref:libraries/base/reference/character_8_chart|CHARACTER]] ,
+[[ref:libraries/base/reference/integer_32_chart|INTEGER]] , and [[ref:libraries/base/reference/real_32_chart|REAL]] are defined by classes of the Kernel library. <br/>
+In reading the class specifications for the numeric types
+[[ref:libraries/base/reference/integer_32_chart|INTEGER]] , and [[ref:libraries/base/reference/real_32_chart|REAL]] you might think that the type declarations are too restrictive. For example the addition operation in class [[ref:libraries/base/reference/real_32_chart|REAL]]  reads 
 <code>
-    infix "+" (other: REAL): REAL
+    plus alias "+" (other: REAL): REAL
 </code>
 
-but there is actually no problem here. A language convention applicable to all arithmetic expressions, the Balancing rule, states that in any such expression all operands are considered to be converted to the heaviest type, where [[ref:libraries/base/reference/real_64_chart|DOUBLE]]  is heavier than [[ref:libraries/base/reference/real_32_chart|REAL]]  and [[ref:libraries/base/reference/real_32_chart|REAL]]  is heavier than [[ref:libraries/base/reference/integer_32_chart|INTEGER]] . So mixed-type arithmetic, consistent with common practice, is possible and indeed frequent. 
+but there is actually no problem here. A language convention applicable to all arithmetic expressions, the Balancing rule, states that in any such expression all operands are considered to be converted to the heaviest type, where [[ref:libraries/base/reference/real_32_chart|REAL]]  is heavier than [[ref:libraries/base/reference/integer_32_chart|INTEGER]] . So mixed-type arithmetic, consistent with common practice, is possible and indeed frequent. 
```

Comment:
Actually this is wrong now. There is no balancing rules anymore, it is controlled by conversions.